### PR TITLE
update create a did + kotlin

### PR DESCRIPTION
**Pull request by acekyd**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [x] 📝 Documentation Update
- [x] 🧑‍💻 Code Snippet

## Description

This PR adds and updates kotlin snippets to Create a DID. 

## Related Tickets & Documents

Resolves #

## Added code snippets?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 general documentation
- [ ] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206268818054117

### CODE REVIEW

File: site/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import web5.sdk.dids.methods.key.DidKey
+:snippet-end:
+
+:snippet-start: createDidDhtKt
+import web5.sdk.dids.methods.dht.CreateDidDhtOptions
+
+//Creates a DID using the DHT method and publishes the DID Document to the DHT
+val keyManager = InMemoryKeyManager()
+val options = CreateDidDhtOptions(
+    publish = true
```
Comment (angiejones) [inline site/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.kt:15]: can you put this all one one line? it's short enough

**Review by angiejones — APPROVED** [review]

File: site/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.kt (hunk 1 above)
Comment (acekyd) [reply site/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.kt]: Thanks!

File: site/docs/web5/build/decentralized-identifiers/how-to-create-did.mdx [hunk 2]
```diff
@@ -3,33 +3,58 @@ title: Create a DID
...
 You can create a DID using any of the [Web5-supported DID methods](/docs/web5/learn/decentralized-identifiers#methods) by using the [DID package](https://www.npmjs.com/package/@web5/dids).
 
 **Import the DID Method you'd like to use:**
 
-```js
+<Shnip
+  snippets={[
+    { snippetContent: importDidsKt, language: 'Kotlin' },
+    ]}
+    inlineSnippets={[
```
Comment (angiejones) [inline site/docs/web5/build/decentralized-identifiers/how-to-create-did.mdx:29]: how come unable to use external snippet here?

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    //DID string
+    val did = didKey.uri
+
+    //DID Document
+    val didDocument = portableDID.didDocument
+    // :snippet-end:
+
+    // Make sure the contents of the VC are as expected
+    assertEquals("1","1")
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:57]: can we add a real test now? just make sure did, portableDID, did, and didDocument aren't null

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import web5.sdk.dids.methods.key.DidKey
+:snippet-end:
+
+/**
+ * Tests backing the Create a DID Guide
+ */
+internal class HowToCreateDidTest {
+
+  @Test
+  fun `createDidKt works`() {
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:24]: change to something less generic like "able to create DIDs"

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    val portableDID = DidDht.resolve(didDht.uri)
+
+    //DID string
+    val did = didDht.uri
+
+    //DID Document
+    val didDocument = portableDID.didDocument
+    // :snippet-end:
+
+    // :snippet-start: createDidKeyKt
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:41]: can we separate these into 2 separate functions?

**Review by angiejones — CHANGES_REQUESTED** [review]

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt (hunk 5 above)
Comment (acekyd) [reply site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:41]: On it

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt (hunk 3 above)
Comment (acekyd) [reply site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:57]: Oh damn! I forgot I pushed that. Was debugging why the test was failing. But yes we can do a real test now.

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js [hunk 6]
```diff
@@ -30,6 +30,19 @@ describe('how-to-create-did', () => {
...
+    const requiredImports = `
+      // :snippet-start: requiredDidImports
+      //did:dht
+      import { DidDhtMethod } from '@web5/dids'
+
+      //did:key
+      import { DidKeyMethod } from '@web5/dids'
+      // :snippet-end:
+    `
+    expect(requiredImports).toBeDefined();
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js:43]: can remove test for imports

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js [hunk 7]
```diff
@@ -30,6 +30,19 @@ describe('how-to-create-did', () => {
     });
   });
 
+  test('show required imports to create did', async () => {
+    const requiredImports = `
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js:34]:
```suggestion

```

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js [hunk 8]
```diff
@@ -30,6 +30,19 @@ describe('how-to-create-did', () => {
...
+      // :snippet-start: requiredDidImports
+      //did:dht
+      import { DidDhtMethod } from '@web5/dids'
+
+      //did:key
+      import { DidKeyMethod } from '@web5/dids'
+      // :snippet-end:
+    `
```
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js:42]:
```suggestion
    
```

File: site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js (hunk 6 above)
Comment (angiejones) [inline site/testsuites/testsuite-javascript/__tests__/web5/build/decentralized-identifiers/how-to-create-did.test.js:43]:
```suggestion
   
```

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 9]
```diff
@@ -0,0 +1,89 @@
+package website.tbd.developer.site.docs.web5.build.decentralizedidentifiers;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions.*
+
+// key manager
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:6]:
```suggestion

```

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+import web5.sdk.crypto.InMemoryKeyManager
+
+//did:dht
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:9]:
```suggestion

```

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+import web5.sdk.dids.methods.dht.DidDht
+import web5.sdk.dids.methods.dht.CreateDidDhtOptions
+
+//did:key
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:13]:
```suggestion

```

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+    //DID string
+    val did = didDht.uri
+
+    //DID Document
+    val didDocument = portableDID.didDocument
+    // :snippet-end:
+
+    assertNotNull(did, "DID should not be null")
+    assertTrue(did.startsWith("did:"), "DID should start with 'did:'")
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:40]:
```suggestion
    assertTrue(did.startsWith("did:dht:"), "DID should start with 'did:dht'")
```

File: site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+    //DID and its associated data which can be exported and used in different contexts/apps
+    val portableDID = DidDht.resolve(didDht.uri)
+
+    //DID string
+    val did = didDht.uri
+
+    //DID Document
+    val didDocument = portableDID.didDocument
+    // :snippet-end:
+
```
Comment (angiejones) [inline site/testsuites/testsuite-kotlin/src/test/kotlin/docs/web5/build/decentralizedidentifiers/HowToCreateDidTest.kt:38]: any way to get cryptographic keys associated with DID in Kotlin?

**Review by angiejones — APPROVED** [review]
